src/Features/CSharp/Portable/PopulateSwitch/CSharpPopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.PopulateSwitch), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal class CSharpPopulateSwitchCodeFixProvider : AbstractPopulateSwitchCodeFixProvider<SwitchStatementSyntax>
+    {
+        protected override SyntaxNode GetSwitchStatementNode(SyntaxNode root, TextSpan span)
```

Comment:
Similarly, if you make AbstractPopulateSwitchCodeFixProvider into ```AbstractPopulateSwitchCodeFixProvider<TSwitchBlockSyntax, TExpressionSyntax>``` then a lot of the abstract methods it uses can be strongly typed to return TExpressionSyntax instead of SyntaxNode.   in the derived types (which will derive from ```AbstractPopulateSwitchCodeFixProvider<..., ExpressionSyntax>```, these will then become strongly typed to return ExpressionSyntax as well.